### PR TITLE
Metadata: add tags field

### LINKED ISSUE #25741: Add a tags field to the metadata files and registry
## Overview
We would like to have a tags field in the metadata of a connector to enable easy searching of the connector registry

https://github.com/airbytehq/airbyte/issues/25139

## Proposed solution
1. Add an optional `tags` field to metadata which is an arbitrary array of strings
2. Copy these tags fields to the final connector registry

### LINKED ISSUE #26204: Fix: Convert supportUrl to documentation url
## Overview
The metadata has a field called `supportUrl` it should be `documentationUrl`

Note we should still support the `supportUrl`

### PR DESCRIPTION
**Pull request by bnchrch**
## What
1. adds an optional `tags` field to the metadata type and registry
2. renames the supportUrl field to documentationUrl

closes #26204 
closes #25741

### CODE REVIEW

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorRegistryDestinationDefinition.yaml [hunk 1]
```diff
@@ -47,6 +47,11 @@ properties:
     description: The date when this connector was first released, in yyyy-mm-dd format.
     type: string
     format: date
+  tags:
+    type: array
+    description: "An array of tags that describe the connector. E.g: language:python, keyword:rds, etc."
+    items:
+      type: string
```
Comment (evantahler) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorRegistryDestinationDefinition.yaml:54]: do you want to enforce the "key:value" thing?  Perhaps a regular expression can be used for that.

File: airbyte-integrations/connectors/source-fullstory/metadata.yaml [hunk 2]
```diff
@@ -14,5 +14,8 @@ data:
     oss:
       enabled: true
   releaseStage: alpha
-  supportUrl: https://docs.airbyte.com/integrations/sources/fullstory
+  documentationUrl: https://docs.airbyte.com/integrations/sources/fullstory
+  tags:
+    - language:low-code
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-fullstory/metadata.yaml:19]: Is low-code a language?  Maybe "low-code:true"?

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorMetadataDefinitionV0.py [hunk 3]
```diff
@@ -148,6 +148,10 @@ class Data(BaseModel):
         "api", "database", "file", "custom", "message_queue", "unknown"
     ]
     releaseStage: Literal["alpha", "beta", "generally_available", "source"]
+    tags: Optional[List[str]] = Field(
```
Comment (evantahler) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorMetadataDefinitionV0.py:151]: Are there required tags?  At least maybe "language:*"?

**Review by evantahler — APPROVED** [review]
Nice!

Some nits, and maybe they can be addressed in future work.  Nothign blocking!

**Comment by evantahler** [comment]
cc @alex-gron - we've got an authoritative place for connector tags (like language) coming soon!

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorMetadataDefinitionV0.py (hunk 3 above)
Comment (bnchrch) [reply airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorMetadataDefinitionV0.py:151]: Evan you reviewed so fast! Thank you!

I put this up slightly prematurely as I was writing the mandatory language check.

Thats now up in a new commit with tests

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorRegistryDestinationDefinition.yaml (hunk 1 above)
Comment (bnchrch) [reply airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorRegistryDestinationDefinition.yaml:54]: updated!

File: airbyte-integrations/connectors/source-fullstory/metadata.yaml (hunk 2 above)
Comment (bnchrch) [reply airbyte-integrations/connectors/source-fullstory/metadata.yaml:19]: I think lowcode counts as a language tag for now.

If we start programming on these tags then we can haul it into a standalone field

**Review by sherifnada — APPROVED** [review]
neat!

**Review by sherifnada — APPROVED** [review]
neat!

**Pull request merged**